src/ftxui/dom/text.cpp

```diff
@@ -27,80 +28,93 @@ class Text : public Node {
     int y = box_.y_min;
     if (y > box_.y_max)
       return;
-    for (wchar_t c : text_) {
-      const int width = wchar_width(c);
-      if (width >= 1) {
-        if (x > box_.x_max)
-          return;
-        screen.PixelAt(x, y).character = c;
-      } else {
-        screen.PixelAt(x - 1, y).character += c;
-      }
-      x += std::max(width, 0);
+    for (auto cell : Utf8ToGlyphs(text_)) {
+      if (x > box_.x_max)
+        return;
+      screen.PixelAt(x, y).character = cell;
+      ++x;
     }
   }
 
  private:
-  std::wstring text_;
+  std::string text_;
 };
 
 class VText : public Node {
  public:
-  VText(std::wstring text) : text_(text) {
-    for (auto& c : text_)
-      width_ = std::max(width_, wchar_width(c));
+  VText(std::string text) : text_(text) {
+    width_ = string_width(text_) ? 1 : 0;
```

Comment:
```suggestion
    width_{std::min(string_width(text_), 1)}
```
in the member initializer list.